lib/ansible/modules/web_infrastructure/wildfly.py

```diff
@@ -0,0 +1,326 @@
+                     {'deployment': deployment}],
+         'json.pretty': 1}
+    return get_response(request_data)
+
+
+def deploy_disable(result, params):
+    if result.get('outcome') == 'failed':
+        return False
+    elif result.get('outcome') == 'success':
+    	if not module.check_mode:
+            undeploy_result, info = undeploy(params)
+            if info['status'] == 200:
+                return True
+            else:
+                module.fail_json(msg='Undeploy failed', details=info['msg'])
+        return True
+
+
+def deploy_enable(result, params):
+    if result.get('outcome') == 'failed':
+    	if not module.check_mode:
+            deployment_result, info = deploy(params)
+            if info['status'] == 200:
+                return True
+            else:
+                module.fail_json(msg='Deployment failed', details=info['msg'])
+        else:
+            return True
+    elif result.get('outcome') == 'success':
+    	if not module.check_mode:
+            undeploy_result, info = undeploy(params)
+            if info['status'] == 200:
+                deployment_result, info = deploy(params)
+                if info['status'] == 200:
+                    return True
+                else:
+                    module.fail_json(msg='Deployment failed', details=info['msg'])
+        else:
+        	return True
+    else:
+        module.fail_json(msg='Deployment failed', details=info['msg'])
+
+
+def main():
+
+    global module
+
+    module = AnsibleModule(
+    	argument_spec=dict(
+            src=dict(type='str', required=False),
+            deployment=dict(type='str', required=True),
+            url_username=dict(required=True, aliases=['username']),
+            url_password=dict(required=True, aliases=['password'], no_log=True),
+            state=dict(choices=['absent', 'present'], default='present'),
+            mode=dict(choices=['domain', 'standalone'], default='domain'),
+            controller_address=dict(default='localhost:9990'),
+            server_groups=dict(type='str', default='main-server-group'),
+        ),
+        supports_check_mode=True,
+    )
+
+    changed = False
+
+    state = module.params['state']
+    deployment = module.params['deployment']
+
+    if not HAS_JSON:
+        module.fail_json(msg='Cannot import JSON package.')
+    
+    deployed, info = is_deployed(module.params)
+
+    if info['status'] == 200:
+        result = json.loads(deployed.read())
+    elif info['status'] == 500:
+    	result = json.loads(info['body'])
+    else:
+    	module.fail_json(msg=info['status'], details=info['msg'])
+
+    if 'outcome' in result:
+        if state == 'absent':
+            changed = deploy_disable(result, module.params)
+            if changed:
+            	module.exit_json(changed=changed, state='undeployed', name=deployment)
+            else:
+                module.exit_json(changed=changed, state='Nothing to undeploy',
```

Comment:
`Nothing to undeploy` nor `undeployed` are not valid states. You should use only `state` here.